### PR TITLE
[polymake] bump to 4.2 and add patch for sigint

### PR DESCRIPTION
**Pull request by benlorenz**
The sigint patch make sure that the Ctrl+C handler of julia is kept, otherwise julia will terminate immediately when ctrl+c is pressed after polymake_jll was loaded (and initialized).
Switched to archive source.

### CODE REVIEW

**Comment by giordano** [comment]
I recommend this pull request isn't merged for a few days since it depends on GMP

**Comment by benlorenz** [comment]
Ok, that's fine with me.

**Comment by fingolfin** [comment]
Since PR #1739 was merged, this PR here could be "resumed" now. Not sure if that requires a rebase by @benlorenz or if it suffices to close and reopen to ensure it is rebuilt based on latest master?

**Comment by giordano** [comment]
CI of pull requests is run on the merge commit against the target branch

File: P/polymake/build_tarballs.jl [hunk 1]
```diff
@@ -94,9 +94,10 @@ install_license COPYING
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
- MacOS(:x86_64),
- Linux(:x86_64, libc=:glibc, compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+    Platform("x86_64", "linux"; libc="gnu"),
+    Platform("x86_64", "macos"),
 ]
+platforms = expand_cxxstring_abis(platforms)
```
Comment (fingolfin) [inline P/polymake/build_tarballs.jl:100]: @giordano thanks for helping out!

@benlorenz was there a particular reason to only allow the cxx11 strings ABI (note: as far as I understand things, the cxxstring ABI does *not* restrict the C++ standard one can use; i.e. one can use the cxx03 string ABI also with C++11 or C++14 code, and vice versa use the cxx11 string ABI in C++03 code.

Comment (fingolfin) [reply P/polymake/build_tarballs.jl:100]: Indeed, see the blue box under https://juliapackaging.github.io/BinaryBuilder.jl/dev/building/#Expanding-C-string-ABIs-or-libgfortran-versions

Comment (benlorenz) [reply P/polymake/build_tarballs.jl:100]: There is no `libcxxwrap_julia` for the cxx03 ABI: https://github.com/JuliaPackaging/Yggdrasil/blob/4f8c5dba511db6bf93cff312b4b6268e16d94df4/L/libcxxwrap_julia/build_tarballs.jl#L36

Comment (giordano) [reply P/polymake/build_tarballs.jl:100]: That may hopefully change in the future

File: P/polymake/build_tarballs.jl [hunk 2]
```diff
@@ -100,17 +109,17 @@ products = [
...
-    Dependency(PackageSpec(name="Perl_jll", uuid="83958c19-0796-5285-893e-a1267f8ec499", version=v"5.30.3"))
-    Dependency("bliss_jll")
-    Dependency("boost_jll")
-    Dependency("cddlib_jll")
-    Dependency("lrslib_jll")
-    Dependency("normaliz_jll")
+    Dependency("CompilerSupportLibraries_jll"),
+    Dependency("FLINT_jll"),
+    Dependency("GMP_jll"),
+    Dependency("MPFR_jll"),
```
Comment (fingolfin) [inline P/polymake/build_tarballs.jl:115]: This now conflicts with master, so I am afraid a rebase is now necessary after all (or just don't add the commas here, it doesn't seem to be necessary?)

File: P/polymake/build_tarballs.jl (hunk 1 above)
Comment (fingolfin) [reply P/polymake/build_tarballs.jl:100]: Note that a bunch of other packages are using `CompilerABI`, e.g. `CMake`, `libcxxwrap_julia`, `libpolymake_julia`, `libsingular_julia`, and more, so I guess all of those should be adjusted? Of course it's a bit annoying that this then means that people who want to test them locally now have to install dev versions of Julia 1.6, BinaryBuilder, and BinaryBuilderBase...

Comment (giordano) [reply P/polymake/build_tarballs.jl:100]: > Of course it's a bit annoying that this then means that people who want to test them locally now have to install dev versions of Julia 1.6, BinaryBuilder, and BinaryBuilderBase...

I know, but unfortunately in BinaryBuilder now we _heavily_ rely on features present in Julia nightly which makes keeping compatibility with older Julia versions nearly impossible, or too much effort to be worth it.  The compatibility shim now used for the platforms doesn't really work great, I'm not going to spend time on it (my bandwidth is limited...), but anyone is welcome to improve it .

Note that to test one of these builders locally the quick and dirty trick is to locally set
```
platforms = Platform[]
```
and manually build from the command line for whatever platform you're interested to.

**Comment by benlorenz** [comment]
Thanks for your help!

**Pull request merged**